src/builtins/filters/mod.rs

```diff
@@ -10,4 +10,13 @@ pub mod object;
 pub mod string;
 
 /// The filter function type definition
-pub type FilterFn = fn(Value, HashMap<String, Value>) -> Result<Value>;
+pub trait FilterFn {
```

Comment:
I don't really like having the `Fn` bit in the type anymore, do you think `Filter` would be ok? (and same for `Test`). It could maybe be confused with https://doc.rust-lang.org/std/iter/struct.Filter.html but that is a struct and it shouldn't be an issue since people can use `tera::Filter` or `import tera::Filter as TeraFilter` for disambiguation.